lib/elixir/lib/list.ex

```diff
@@ -591,7 +591,12 @@ defmodule List do
   """
   @spec to_string(:unicode.charlist) :: String.t
   def to_string(list) when is_list(list) do
-    case :unicode.characters_to_binary(list) do
+    try do
+       :unicode.characters_to_binary(list)
+    rescue
+      ArgumentError ->
+        raise ArgumentError, "cannot make this list into a string (use 'inspect(list)' for its internal representation as a string)"
```

Comment:
> Do you think mentioning "inspect" is a bad idea? 

Yes, because we are not sure that's the actual solution. Maybe it was supposed to work, maybe it was meant to be a char list, but there was a bug in the code and mentioning inspect will point people to the wrong direction.